### PR TITLE
feat(redis): start redis on server start

### PR DESCRIPTION
**Pull request by udaraweerasinghege**
npm start / pm2 start servers.json will start redis now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "lodash": "4.16.2",
     "minimist": "1.1.1",
     "pm2": "2.2.3",
-    "replace-in-file": "0.2.1"
+    "replace-in-file": "0.2.1",
+    "redis-server": "1.1.0"
```
Comment (vladikoff) [inline package.json]: @udaraweerasinghege does Travis-CI need `redis` installed? 

You might need to add https://docs.travis-ci.com/user/database-setup/#Redis

Comment (vladikoff) [inline package.json]: Also there is a test file used by `.travis.yml` that pings the ports. Can we somehow ping the redis server to check that it started?

Comment (udaraweerasinghege) [inline package.json]: On travis that seems to start the redis server which we don't want. I'll add a test to ping the server

File: _scripts/redis.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const RedisServer = require('redis-server');
```
Comment (udaraweerasinghege) [inline _scripts/redis.js]: Our instructions already ask the user to install redis ```brew install gmp redis graphicsmagick```

File: test/curl.sh [hunk 3]
```diff
@@ -47,3 +62,6 @@ check 127.0.0.1:5050 405
 # sync server
 # address of the endpoint have to be the same as a public_url in settings
 check localhost:5000
+
+# redis server
+check_redis
```
Comment (udaraweerasinghege) [inline test/curl.sh]: @vladikoff added a test to make sure redis is up
Reference:
Using ping : https://stackoverflow.com/questions/21708107/how-do-i-check-if-redis-is-running-before-i-start-flask-and-start-it-if-it-isn

nc for pinging redis : https://stackoverflow.com/a/31195882

File: _scripts/redis.js (hunk 2 above)
Comment (vladikoff) [inline _scripts/redis.js]: @udaraweerasinghege oh lol, that's great!!

**Comment by udaraweerasinghege** [comment]
Hmm this seems to make auth-db server fail to start. Might have to do with order. Investigating.

**Comment by udaraweerasinghege** [comment]
@vladikoff Hmm mine was breaking locally for that so thought it was because it was connecting to redis when it didn't exist. Not sure why travis is breaking. I'll check it out

**Comment by udaraweerasinghege** [comment]
Seems to fail in setting up oauth-console on travis o_o

**Pull request merged**